android/src/main/java/com/pagopa/ioreactnativecie/IoReactNativeCieModule.kt

```diff
@@ -0,0 +1,291 @@
+package com.pagopa.ioreactnativecie
+
+import android.util.Base64
+import android.util.Log
+import com.facebook.react.bridge.Callback
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.pagopa.ioreactnativecie.cie.Atr
+import com.pagopa.ioreactnativecie.cie.CieType
+import it.pagopa.io.app.cie.CieSDK
+import it.pagopa.io.app.cie.cie.CieAtrCallback
+import it.pagopa.io.app.cie.cie.NfcError
+import it.pagopa.io.app.cie.cie.NfcEvent
+import it.pagopa.io.app.cie.network.NetworkCallback
+import it.pagopa.io.app.cie.network.NetworkError
+import it.pagopa.io.app.cie.nfc.NfcEvents
+
+
+class IoReactNativeCieModule(reactContext: ReactApplicationContext) :
+  ReactContextBaseJavaModule(reactContext) {
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  /**
+   * Lazy value ensures it is initialized with a valid activity when first used.
+   */
+  val cieSdk: CieSDK by lazy {
+    CieSDK.withContext(currentActivity)
+  };
+
+  @ReactMethod
+  fun addListener(eventName: String) { }
+
+  @ReactMethod
+  fun removeListeners(count: Int) { }
+
+  @ReactMethod
+  fun hasNfcFeature(promise: Promise) {
+    try {
+      cieSdk.hasNfcFeature().let {
+        promise.resolve(it)
+      }
+    } catch (e: Exception) {
+      ModuleException.UNKNOWN_EXCEPTION.reject(
```

Comment:
If you like, you could type-alias [ModuleException](https://github.com/pagopa/io-react-native-cieid/blob/8331da926a0b68462feefdc5c4262b4aeecbc32e/android/src/main/java/com/ioreactnativecieid/IoReactNativeCieidModule.kt#L19C1-L19C66), similar to what we did [here](https://github.com/pagopa/io-react-native-cie/pull/1/link).

What do you think?

Reply:
Yes absolutely. Added here: https://github.com/pagopa/io-react-native-cie/pull/1/commits/8893643908bfd1f098464a26ffc79658ac812170

Thank you!